my-sms/app/controllers/notes_controller.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: false
+
+class NotesController < ApplicationController
+  before_filter :load_notable
+  before_filter :load_note, except: %i[index]
+
+  def new
+    @notable_instance = @notable_type.classify.constantize.find(@notable_id)
+  end
+
+  def create
+    respond_to do |format|
+      if @note.save
+        format.html { redirect_to @note.notable, notice: 'Note was successfully created.' }
+        format.json { render json: @note, status: :created, location: @note }
+      else
+        # propagate note errors to root model, if possible
+        flash_errors(@note)
+        format.html { redirect_to @note.notable || :back }
+        format.json { render json: @note.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  def update
+    respond_to do |format|
+      if @note.update_attributes(params[:note])
+        format.html { redirect_to @note, notice: 'Note was successfully updated.' }
+        format.json { head :no_content }
+      else
+        format.html { redirect_to @note.notable || :back }
+        format.json { render json: @note.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  def destroy
+    @note.destroy
+
+    respond_to do |format|
+      format.html { redirect_to @note.notable }
+      format.json { head :no_content }
+    end
+  end
+
+  def load_notable
+    @notable_type = params[:notable_type]
+    @notable_id = @notable_type && params["#{@notable_type.downcase}_id"]
+  end
+
+  def load_note
+    @note = if params[:id]
+              Note.find(params[:id])
+            else
+              note = Note.new(params[:note])
+              note[:notable_type] = @notable_type
+              note[:notable_id] = @notable_id
+              note
+    end
+  end
```

Comment:
I still think to refactor most of this file into a concern, and put the notable loading logic into it's own controller. You'll be tripling up on the notes controller, but each controller will look as small as this:

```ruby
class Students::NotesController < ApplicationController
  include Concerns::NotesConcern

  private
  
  def load_notable
    @notable = Student.find(params[:student_id])
  end

end
```

And then in the concern you can remove the `load_notable` method (since that's implemented in each concrete controller) and then your `load_note` method becomes something like


```ruby
def load_note
  @note = @notable.notes.find_or_initialize_by(params[:id])
end
```